### PR TITLE
Add new binding: SWARM Tile module

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- This adds a binding for the [SWARM](https://swarm.space) Tile module.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: devices/SwarmTile/SwarmTile/Commands/GpsFixQualityCmd.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    internal static partial class TileCommands
+    {
+        /// <summary>
+        /// Command to set or query the unsolicited report messages for GPS fix quality.
+        /// </summary>
+        internal class GpsFixQualityCmd : CommandBase
+        {
+            public const string Command = "GS";
+
+            public int Value;
```
Comment (Ellerbach) [inline devices/SwarmTile/SwarmTile/Commands/GpsFixQualityCmd.cs:18]: Need expliciti setter/getter

File: devices/SwarmTile/SwarmTile/Commands/GpsFixQualityCmd.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                public Reply(NmeaSentence sentence)
+                {
+                    try
+                    {
+                        if (sentence.Data.Length >= 6)
+                        {
+                            // $GS <hdop>,<vdop>,<gnss_sats>,<unused>,<fix>*xx
+                            //     |                                                |
+                            //     3
+
```
Comment (Ellerbach) [inline devices/SwarmTile/SwarmTile/Commands/GpsFixQualityCmd.cs:46]: extra line here (linter won't be happy)

File: devices/SwarmTile/SwarmTile/Commands/DateTimeStatus.cs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                            else
+                            {
+                                // all the rest assume invalid
+                                DateTimeInfo.IsValid = false;
+                            }
+                        }
+                        else if (!sentence.Data.Contains("OK"))
+                        {
+                            // must be the current DT rate
+
```
Comment (Ellerbach) [inline devices/SwarmTile/SwarmTile/Commands/DateTimeStatus.cs:92]: extra line for the linter as well

File: devices/SwarmTile/SwarmTile/Commands/Gpio1Control.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    internal static partial class TileCommands
+    {
+        /// <summary>
+        /// Command to set, query and control of the GPIO1 pin to allow indications or control the operation of the Tile.
+        /// </summary>
+        internal class Gpio1Control : CommandBase
+        {
+            public const string Command = "GP";
+
+            public GpioMode Mode;
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply devices/SwarmTile/SwarmTile/Commands/Gpio1Control.cs:18]: Why is that? We are OK with auto-properties if there is no need to use the backing field from native... 🤔

**Comment by josesimoes** [comment]
I've added Stylecop to the project and I got 480+ errors!!! 😱 
This is very very strict.
Fixed several and quite some still to fix.

cc @torbacz

**Comment by torbacz** [comment]
Good luck fixing all of them :D
As for rules, I'm not sure if all of them are correct. They were tested against rather small project, feel free to adjust them if needed (there is a tool inside \packages\StyleCop.MSBuild.6.2.0 folder).

**Comment by Ellerbach** [comment]
@torbacz, I adjusted a bit the rules. So this PR contains better rules, removed few things we're not even using in the .NET IoT repo. So it should be a bit closer now. Thanks again for your work on this!
If you would make the PR with this adjusted file, it would be great! Also adding in the main contribution doc that developers should use this new linter.

@josesimoes I fixed all issues, I adjusted the structure as well, the files were all in a too deep directory.
I had to remove the GitVersionning element from the project. So please please add it back!
Rest should be fine :-)

**Review by Ellerbach — APPROVED** [review]
Thanks for the addition, I think between the linter fix and the previous review we should be good to go

**Pull request merged**